x-pack/plugins/global_search/kibana.json

```diff
@@ -0,0 +1,10 @@
+{
+  "id": "globalSearch",
+  "version": "8.0.0",
+  "kibanaVersion": "kibana",
+  "server": true,
+  "ui": true,
+  "requiredPlugins": ["licensing"],
```

Comment:
So, I've added `licensing` as a required dep but I'm ensure exactly how I should use it.

As GS, as a plugin, is not a user-facing feature but just an API to be consumed by other plugins, I don't know what we should do when the license is invalid/expired. (This was something that we missed in the RFC as the licensing decision came later)

One option would be to degrade the `find` API when the license is not in a valid state, either by returning no results, or by throwing an error (either from `find` itself or from the returned observable). Note that I find none of these possibility very good, as:
- returning no results would not allow consumers to know that the empty result response was caused before of license check
- throwing an error feels slightly better (I would go with the observer error in that case), but I'm afraid that checking for such errors would be forgotten by consumers, resulting in actual uncatched runtime errors on either client or server side.

Another option would be to decide that only user-facing features using GS (such as the searchbar) should perform such verification, and that the GS API should not check for license state.

WDYT @joshdover @kobelb @ryankeairns ?

Reply:
> throwing an error feels slightly better (I would go with the observer error in that case), but I'm afraid that checking for such errors would be forgotten by consumers, resulting in actual uncatched runtime errors on either client or server side.

I'd prefer this approach. This would behave similarly to our HTTP APIs which are used when the license is expired or is invalid. There is the risk that consumers won't handle the error, but worst case we end up with an unhandled exception, which makes trouble-shooting the lack of proper error handling much easier than just returning no results.

Reply:
License checking added to the `find` API